packages/components/modal/tests/dialog.test.tsx

```diff
@@ -0,0 +1,106 @@
+import { Button, useDisclosure } from "@yamada-ui/react"
+import { a11y, fireEvent, render, screen } from "@yamada-ui/test"
+import { Dialog, DialogHeader, DialogFooter, DialogBody } from "../src"
+
+describe("<Dialog />", () => {
+  const DialogOpenExample = () => {
+    const { isOpen, onOpen, onClose } = useDisclosure()
+
+    return (
+      <>
+        <Button data-testid="OpenDialog" onClick={onOpen}>
+          Open Dialog
+        </Button>
+
+        <Dialog
+          data-testid="Dialog"
+          isOpen={isOpen}
+          onClose={onClose}
+          cancel={{
+            children: "dialog-cancel",
+          }}
+          onCancel={onClose}
+          other={{
+            children: "dialog-other",
+          }}
+          onOther={onClose}
+          success={{
+            children: "dialog-success",
+          }}
+          onSuccess={onClose}
+        >
+          <DialogHeader data-testid="DialogHeader">header</DialogHeader>
+          <DialogBody data-testid="DialogBody">body</DialogBody>
+          <DialogFooter data-testid="DialogFooter">footer</DialogFooter>
+        </Dialog>
+      </>
+    )
+  }
+
+  const DialogCloseExample = () => {
+    const { isOpen, onOpen, onClose } = useDisclosure()
+
+    return (
+      <>
+        <Button data-testid="OpenDialog" onClick={onOpen}>
+          Open Dialog
+        </Button>
+
+        <Dialog
+          data-testid="Dialog"
+          isOpen={isOpen}
+          onClose={onClose}
+          cancel={{
+            children: "dialog-cancel",
+          }}
+          onCancel={onClose}
+          other={{
+            children: "dialog-other",
+          }}
+          onOther={onClose}
+          success={{
+            children: "dialog-success",
+          }}
+          onSuccess={onClose}
+        ></Dialog>
+      </>
+    )
+  }
+
+  test("Dialog renders correctly", async () => {
+    const { container } = render(<DialogOpenExample />)
+    await a11y(container)
+  })
+
+  test("Dialog renders correctly when open", async () => {
+    render(<DialogOpenExample />)
+
+    expect(screen.getByTestId("OpenDialog")).toBeInTheDocument()
+
+    fireEvent.click(screen.getByTestId("OpenDialog"))
```

Comment:
Unless there is a specific reason not to, you should use `user-event` instead of `fireEvent`. 
It allows you to conduct tests in scenarios that are closer to actual user interactions.

Please see [this page](https://ph-fritsche.github.io/blog/post/why-userevent).



```ts
test("Dialog renders correctly when open", async () => {
  const { user } = render(<DialogOpenExample />)

  expect(await screen.findByTestId("OpenDialog")).toBeInTheDocument()

  await user.click(screen.findByTestId("OpenDialog"))
  
  expect(await screen.findByTestId("Dialog")).toBeInTheDocument()
  expect(await screen.findByTestId("DialogHeader")).toHaveTextContent("header")
  expect(await screen.findByTestId("DialogBody")).toHaveTextContent("body")
  expect(await screen.findByTestId("DialogFooter")).toHaveTextContent("footer")
})
```

Reply:
Thanks, I have fixed.